adoc/chapters/programming_interface.adoc

```diff
@@ -5186,718 +5173,637 @@ If no template argument is provided, then the default allocator for the SYCL
+buffer(T* hostData, const range<Dimensions>& bufferRange,          (1)
+       const property_list& propList = {});
 
-Zero or more properties can be provided to the constructed SYCL
-[code]#buffer# via an instance of
-[code]#property_list#.
+buffer(const T* hostData, const range<Dimensions>& bufferRange,    (3)
+       AllocatorT allocator, const property_list& propList = {});
 
-a@
-[source]
+buffer(const T* hostData, const range<Dimensions>& bufferRange,    (4)
+       const property_list& propList = {});
 ----
-buffer(const T* hostData,
-       const range<Dimensions>& bufferRange,
-       AllocatorT allocator,
-       const property_list& propList = {})
-----
-   a@ Construct a SYCL [code]#buffer# instance with the [code]#hostData#
-      parameter provided.  The buffer assumes exclusive access to this
-      memory for the duration of its lifetime.
-
-The constructed SYCL [code]#buffer# will use the
-[code]#allocator# parameter provided when allocating
-memory on the host.
-
-The host address is [code]#const T#, so the host accesses
-can be read-only. However, the [code]#typename T# is not const so
-the device accesses can be both read and write accesses. Since,
-the [code]#hostData# is const, this buffer is only initialized
-with this memory and there is no write back after its
-destruction, unless the [code]#buffer# has another valid
-non-null final data address specified via the member function
-[code]#set_final_data()# after construction of the
+_Effects (1):_ Construct a SYCL [code]#buffer# instance with the
+[code]#hostData# parameter provided.
+The buffer is initialized with the memory specified by [code]#hostData#, and the
+buffer assumes exclusive access to this memory for the duration of its lifetime.
+The constructed SYCL [code]#buffer# will use the [code]#allocator# parameter
+provided when allocating memory on the host.
+The range of the constructed SYCL [code]#buffer# is specified by the
+[code]#bufferRange# parameter provided.
+Zero or more properties can be provided to the constructed SYCL [code]#buffer#
+via an instance of [code]#property_list#.
+
+_Effects (2):_ Equivalent to `buffer(hostData, bufferRange, AllocatorT{},
+propList)`.
+
+_Effects (3):_ Construct a SYCL [code]#buffer# instance with the
+[code]#hostData# parameter provided.
+The buffer assumes exclusive access to this memory for the duration of its
+lifetime.
+
+The constructed SYCL [code]#buffer# will use the [code]#allocator# parameter
+provided when allocating memory on the host.
+
+The host address is [code]#const T#, so the host accesses can be read-only.
+However, the [code]#typename T# is not const so the device accesses can be both
+read and write accesses.
+Since, the [code]#hostData# is const, this buffer is only initialized with this
+memory and there is no write back after its destruction, unless the
+[code]#buffer# has another valid non-null final data address specified via the
+member function [code]#set_final_data()# after construction of the
```

Comment:
I don't think the bit about the host vs device access adds very much here.  We could keep it as non-normative, but I think it's likely to confuse people more than it helps -- we shouldn't have to explain what `const` means.

I think replacing this entire paragraph with the below would be a lot clearer:

```suggestion
_Remarks_: The buffer will not write back to any host memory
unless requested via the member function [code]#set_final_data()#.
```

Reply:
Remarks seem to be a separate segment of the description, so to avoid having to associate it with the constructors by their callouts, I added a few more words to the beginning.